### PR TITLE
#175537347 Implement CRUD APIs for food

### PR DESCRIPTION
**Pull request by Extremophile-ai**
## What does this PR do? ##
- Enables a user with admin or super admin role to create a food.
- Enables a user to get all Foods
- Enables a user to get all foods of a country
- Enables a user to get all one food
- Enables admin to update an existing food
- Enables admin to delete an existing food
## Description of tasks to be completed ##
- Create a controller to create food under an existing country
- Create a controller to get all foods in Africa
- Create a controller to delete a food with a valid Id
- Create a controller to update an existing food

- Create a `POST` route to create food by country
- Create a `GET` route to get all foods
- Create a `GET` route to get a single food with a valid Id
- Create a `GET` route to get all foods of a country
- Create a `DELETE` route to delete a food with a valid Id
- Create a `PATCH` route to update an existing food
- Write tests for the controllers and routes
## How should this be manually tested/checked? ##
- Clone the PR
- Checkout to the branch ft-user-can-get-food-175537347
- Install the dependencies using npm install
- Create a .env file and define the following as follows:
- DEV_DATABASE_URL = <development database url>
- TEST_DATABASE_URL = <test database url>
- DATABASE_URL = <production database url>
- run npm run migrate to create the database tables
- run npm run seed to seed the database
- Start the server by running npm run start-dev.

#### Sign in a User with admin role ####
Sign in via `http://localhost:3000/api/v1/users/signin` using a POST method with
```
        {
             "email: "ufuoma@gmail.com",
             "password": 12345
        }
```
and copy the token (without the quotes) from the response body.
NOTE: To create, update, or delete a food, set the Authorization to Bearer Token and paste the copied token in the space provided.

#### Create a Food ####
To create a new food, make a POST request to `http://localhost:3000/api/v1/admin/food/2e11e4a9-441b-4426-9521-39adc64ccfad` and add the following to the body:
```
{
    "foodName": "Nshima",
    "methodOfPreparation": "Nshima is the staple carbohydrate of Zambia. It is made from corn that is processed into a fine white powder called ‘mealie meal’. It is cooked by mixing the corn meal with water which is brought to the boil as porridge (similar to grits). More cornmeal is added until it develops a thicker texture. The cornmeal can be substituted for cassava, sorghum and millet. Nshima is served with a protein (usually meat or fish) and one or two vegetables. Nshima can be enjoyed at any of the many traditional restaurants in Zambia as well as at boutique hotels. ",
    "gallery":"https://images.unsplash.com/photo-1482049016688-2d3e1b311543?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=500&q=60"
}
```
#### Get foods ####
To get all foods in Africa, make a `GET` request to `http://localhost:3000/api/v1/food`.
To get a single existing food, make a `GET` request to `http://localhost:3000/api/v1/food/9e92fadd-8897-4d3d-b9de-cf82e9752a1f`.
#### Update Food ####
To update an existing food, make a PATCH request to `http://localhost:3000/api/v1/admin/food/9e92fadd-8897-4d3d-b9de-cf82e9752a1f` and place the fields to be updated in the request body.
request body sample:
```
{
    "foodName": "Egusi stew"
}
```
#### Delete Food ####
To delete an existing food, make a DELETE request to `http://localhost:3000/api/v1/admin/food/9e92fadd-8897-4d3d-b9de-cf82e9752a1f` 

## Architectural Flow chart ##
[Food flowchart](https://viewer.diagrams.net/?highlight=0000ff&edit=_blank&layers=1&nav=1&title=Food%20flowchart.drawio#R5V1bV%2BLKEv41PDIr98sjo%2Bh2LUfd6uzZPjIShbWAuBFHPL%2F%2BdEg6JF0VAk13p0leFHIpkrp119f1JT37bL6%2BXI7eJj%2FicTTrWcZ43bPPe5bl2HZA%2FiVbvtItpmVlW16X03G2bbvhYfq%2FKNtoZFs%2FpuPovXTgKo5nq%2BlbeeNzvFhEz6vSttFyGX%2BWD3uJZ%2BVffRu9RmDDw%2FNoBrf%2Bmo5Xk2yraRjbHX9F09dJ9tOBm%2B2Yj%2BjB2Yb3yWgcfxY22cOefbaM41X6ab4%2Bi2aJ9qhe0vMuKvbmF7aMFqt9Tpi%2Fjmbrs%2Fjzen3xtrDGl%2Fbt1V%2F9TMqf0ewju%2BGe5c2IvO8vMRFLrnr1lanC%2B%2B8jpjv67xtDDcgBlvO23u4kn16T%2F2f3w8HjkOy%2BGf4ify9ub8%2BpYHKFqez0yEw5%2Bc9Yq2idbJ%2Bs5jOywSQfR7Pp64J8fiZ3Gi3Jhj%2FRcjUlNhpkO%2BbT8Tg5%2FfsyIhc2%2Br0RZZDvb%2FF0sdp4gPu9554nsj5WcXrxG9FQiVQj5CeidWFTptTLKJ5Hq%2BUXOSTb26eumrk4tffn1l0s6hSTgqfQ40aZh77mkrdGJB8yOx5gUxvY9OFxcP8IFE1EkShK1PY5ma6ih7fRc7Lnk0RyWf8ClOSWdWSaUEn5tqKSXFlKchDHZ%2FWzGA%2BSDJJ43mz0%2Fj593lMt0biUU6BSCneNeQbdtoxmo9X0TzkTYZrIfuEu8fatzk3D%2BeaW1e4x6nyPP5bPUXZeMXfUisoTNBW1Gi1foxUQtTFOfuv89nKBvYb397f3wGgky74lH4lZRrNZNItfl6N5kgmi5ZRcQ5I%2Byvvutjvq4uBluo7oECUoLvIskKnVtlwQGKHKuPCAnq8eyPeHq8ubIflmXN307Aug9eUknv%2F%2BeFeSShyfcUV732QSyFKaDzMu0VeqLaCq%2BGMxjsbZAKVAXRVxW1KWC5XlyFJW0IXMa5WVzobr%2Fnm3LMhWnHXDLtiKHdrsUNgo6bADrmR7UfcoGOxp%2BABtRlR%2FPfpNajVp82xo9p0OxqazvKDLfqRXrJnQWbjxzSmpnpY2vO5BD4lfXt4jOcbCiq%2B2RZfHzG94M6Fl1wiSHVkWMNbNrcaBZVZNFHgiy3WssKR9%2FWMLFsGti618PpvXDrzjFhOkeazVxBbR3uircFjmrHtfMP2dre1TiWI9AVb6m4pmcP5Dg1qmYrpQnJxbKisZE9bZrYsbVulhSGdthwZO364VJXtYgvW63sMS9S8Rw5Kt%2FSgEgYFWoFZ9k4Gt0MRlICHM1kPiEpcAWCFaT1f%2FJh7%2BzfWzr0%2FkK%2FE0w6QbztdZCGy%2BfBW%2BFEySbluQ%2B%2Fq3%2BOUplU2%2FbkVtvlFZlfZJs9Ie7pbmnB0H%2Blrl474X4s50aDpmoS52PiRoGtP37PLvKJnGdBKKcVkUcn8opizIM9SWi9T1TgaIMUUiMW4YhrqPzRZEX9JVauN8%2BDi4uobG2iL3ZhPFQe7CdcWBtDGWppx25yCrrHX%2BJdOyIF91BoIIyD%2BD66vztC1jb1dvZJEqbN7VO9kfEIrrDyDur9jfIXKxwXnObn%2FePN4%2FkU8b9yf%2FmwZ9LHYB28fcHTO8yQJz4vwd4ggH%2BztvwZOXXYWaKzursuCqNENtkUT9pLZI0ixagdcEgcONW%2FlGnahjayXq2p6LXwnd7xo793sqaisLwjRag2h5sIqYq1uhVS5nj1zbWaMnSJzHd7CfJGDHgQMmpeyQr3paCnEMvQvj3L%2BEINam45XU37eOcxD58WVDJIN%2FZmDyzQxMJTMD2zrNmYHpMSHNC5pZJjPDYBuhRK0DG%2BXph2l4FbOAihPoHe57vJJZhA0BJa1nEXlgC5lFsIBrX%2FvFORrubMGYlokN14geO05v6%2BlCUgmQpCKtNcDuQEsNO9Ha8ryOhf9MkzWM5ImWDREsvSdauX%2BJ7wXVvl3NhuBVzqBLoVo2FzUL0G69uTGE1haAWOmfjpiJEmBl7J%2BOWEl7zu2EeTjCkXm8vYMmU0hKZJ0610mJG5OPxKLd2lp%2FPF3%2FGq5%2FPf5992di%2FnPxfPZF60HhhNyfd6WVH22puEj2BzauNKjrM3k%2FgPYMLGhNmjqEWxOOwUKouEcpyWP4ykq5uKiWhPWcMml7L62oyeSeJ4yLC0VJ5OKi9oIjry5djUcFhg%2F66dVxcVE9wyFTAhf3uFxiNErGRbUG1wOEknGP1FeDbFxUW8LaCnXOveVSnJuNywqSyMZFbYWwO1toLGF0XChKIh0XNxiEhBvFYPaGVsoOscMZ98dgtOLj4neE1V4tCy%2Bffd4Iby70wxpBskMLotFNrrYIjSyrLrK05uPi9ySsn1Lf2Aoctn7gjS0mSGXxcdkLFk1kwT2hok9TFB9X6PxcKSEXV5cwmFvfwAFa52fkulatKNkDk17NhEIHJq9uYNJp0R%2B%2FBYgOtAK7cphUrpSQi2taALKgEyEXpM9iP9lOXyv2k%2B1CrLTJx05o4s50cAXOwl2SGLlO6Jd%2FR8VEBqF4tm9kBhAKNyPXY7oUJTJycXO1GIs5OUYufhcQf8loinTx2hvNk6F08fv9baPHLX%2FRGNwkB2bHX50Du3KQd4VWEkrZu7h6hfXv6ZyvrLLW%2BRdZy4IksndxY0G8BLJ3854OgTTeI52%2Bkl7SnNt3sruAm8mLNCpIZPLiFqt4BnUhuYvk8h7XcRQ0S%2BbFFQghiINdXj2Z9%2FACizpKbYFF71%2BbgIVucwSb12GWYiSweellMw%2BKrGT3Vh1fxfatOF40bwd3DggJtQWwSxPBrrrghNi%2F%2BB12sH%2BFm%2F3rAVaRRPYvai6ETNqeIjysCza92b%2B4wWDRKJQkd1y5YTXKksMV1oGFbjYh8bPkoCR5LDncXLA6bE1CSl1xV0LSlyWH3xAsDBlaS%2BP4R5M0OVxnAmpB%2FfORKJoclCSPJoebC%2BleF0CTE%2BvVOvDkbBHLy8ofS6JvDPkAus2fysHRv8asfUjqX%2FPZaK17jgh7Qt1zRNjjleARDqyR2oJH2LUlkt7PEcHtBReWmyd59h32jatIYaSW5elYx2ds3bMo8XCXTaM%2BZxZFZO3bPyPMtWH134puuT54dRKtA5uiejoQNtCN60lSCjtFsPdNKtLgFgfW7%2FqwPcnVVcRvY3RPpwPt5CRxBmW1805kgSRA6ZCdgDtQvSPjnMOLJ2Ky2PFXtslgBd8aRNGpXU88Nc6n04H1w77NEtO486Hl10iSHFxUIy2sjZ3a2vj0WJ8uLI3bF125ifNagnvwYvmY7DRQ1AvM2Eu2yjiYHFzLhfW%2FPsRPMFXPpw2NMT%2FdDvRrA7W7Bv%2FLWF27Vpbs4QlW8W0ZnlJv3DU8aQ%2FUuhAraAea5THpHM1dSsmfrojXCZ00%2BZM6W21vMkWw9MnJLvs%2BVm4EG%2BBfkhYCySUzr2RVMqHpJjjj8j4yKJFVkuQpfn6a22Joxq2FZk6BAupCOIYyffShdoK05mFPcVTa5UUFtzwVlcF474iV1bIkX%2FUDdCu6qnVkuPUdn1328zF%2FV0tx8wS0WZ8CxY16Su00kmZOfcIV%2Bk0Q8r%2Bx0mVfWQllCZtMMu9K8P3dLDd4Qk0bGjjBUzFd9WD52xZwwqvtdD91npvXxXX7gLeNm0hi%2BbWh6iEeFoetqTW8%2BufAnR7TzYdrVXqRS0jZUenUjRUevtWBvOTAx5rwQ2JIn73ipiIfVh%2BNc0wQ50a5nKppJj427op4H9P58HrY%2Bvcx9Z2AsSn2QiYkctmIEGdPDLMVYc%2FL4eO2gP%2F%2BRP6c3f68ebx%2Faq1tTYOpYpBwtbFVJ3nGxchzjTMxggBk%2FMa5GH4X2t4Cq1xO8DMxWEmqeRgBnJvm2cW4GfwYAuMpx8QD8EQjtNld6eSUvupPy4dXIHkBY6EofndWF3qOQDSz6uTOC%2FaeDXvC8kI7UHDkJa61wHga2835EEh3Nr8XwdS558t0Doa2QxvkHL8Gq4an0KuTilYH7UWrg9aj1UEXOmPY5O%2BYvGtbQFKgtm01aHFfTNA6ylIAa7cUfBhcX%2FdSCCKZ5N5eJJt6BRyiskRQhF%2FDcc7BHiSjtESgFyAJDxr0Wg7wmRb7DBYE4fMQi0oDgUJYKzePbsNaD30fgGJwO4T1cfuHaf72VVaS6jeYhNi6TePwpmmZuqGboYCyWHu%2FNklaFfaoGUSYaowzhAWfRi2%2FJqBMNo5uhhWvSdAD3QRpoXlwM%2BxCVYpFMjcyhQhTDXGGsDY92GgaQJy1eCaNZo08CRDb%2Bf0IZE9ZACepRw7EN8EZSuDNEBbxbYE304htM7yZc2m7NpLwQ5yYMMUop2lAZKA1MGfmkm3COU0DwhMpqvZwNzy7urgiYksNVzvrBEUYJxjpmoc4TUPHfk9QH%2BgAiJlGR5EDflAMEaYaF8sfGS0LxC%2Buq7QVx2f5dfT17QWftJQ2c5oGxDd0gDtdJsE3D3eaRieADccT92htRJhqvDN%2F7u%2FuddzGJzSuVaGn5iY0JpYZGp%2FQsIkhfw5gkxMas6Op4QjMEwoDT3TkTg3k6zJOphjbw4kzTn7E46TcHP4f)
## What are the relevant pivotal tracker stories? ##
[#175537347](https://www.pivotaltracker.com/n/projects/2463429/stories/175537347)

### CODE REVIEW

**Comment by bellogo** [comment]
When adding food, could you please put the country name in the req.params instead of the req.body

**Review by Fiyin-Anne — APPROVED** [review]

**Review by Fiyin-Anne — APPROVED** [review]

**Review by bellogo — DISMISSED** [review]

**Review by Fiyin-Anne — DISMISSED** [review]

**Review by Fiyin-Anne — DISMISSED** [review]

**Review by bellogo — APPROVED** [review]

File: src/controllers/food/food.js [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+export default class Food {
+  /**
+   * @param {object} req - The createFood request object
+   * @param {object} res - The createFood response object
+   * @returns {object} Success message
+   */
+  static async createFood(req, res) {
+    try {
+      const {
+        country,
```
Comment (codeliezel) [inline src/controllers/food/food.js:22]: Please use the ID instead.

File: src/routes/foodRoute/foodRoutes.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  updateFood,
+  deleteFood
+} = Food;
+const { verifyAdmin, verifyToken } = Authentication;
+const router = Router();
+router.post("/admin/food", verifyToken, verifyAdmin, createFood);
+router.patch("/admin/food/:id", verifyToken, verifyAdmin, updateFood);
+router.delete("/admin/food/:id", verifyToken, verifyAdmin, deleteFood);
+router.get("/food", getAllFoods);
+router.get("/food/country-foods/:country", getFoodsByCountry);
```
Comment (codeliezel) [inline src/routes/foodRoute/foodRoutes.js:19]: Just link it the way your tms did.

File: src/routes/foodRoute/foodRoutes.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  createFood,
+  getFoodsByCountry,
+  getAllFoods,
+  getOneFoodById,
+  updateFood,
+  deleteFood
+} = Food;
+const { verifyAdmin, verifyToken } = Authentication;
+const router = Router();
+router.post("/admin/food", verifyToken, verifyAdmin, createFood);
```
Comment (codeliezel) [inline src/routes/foodRoute/foodRoutes.js:15]: with the countryID

File: src/services/AdminServices/countryService.js [hunk 4]
```diff
@@ -29,4 +29,36 @@ export default class Admin {
...
+  static async checkCountryById(countryId) {
+    try {
+      return await database.Countries.findOne({ where: { id: countryId } });
+    } catch (err) {
+      throw err;
+    }
+  }
+
+  /**
+=======
```
Comment (codeliezel) [inline src/services/AdminServices/countryService.js:47]: you have conflict markers here

**Review by codeliezel — CHANGES_REQUESTED** [review]
Rework those reviews

File: src/controllers/food/food.js (hunk 1 above)
Comment (Extremophile-ai) [reply src/controllers/food/food.js]: Done

File: src/routes/foodRoute/foodRoutes.js (hunk 3 above)
Comment (Extremophile-ai) [reply src/routes/foodRoute/foodRoutes.js]: done

File: src/services/AdminServices/countryService.js (hunk 4 above)
Comment (Extremophile-ai) [reply src/services/AdminServices/countryService.js]: fixed

**Review by bellogo — APPROVED** [review]

**Review by Fiyin-Anne — APPROVED** [review]

**Review by bellogo — APPROVED** [review]

**Review by Fiyin-Anne — APPROVED** [review]

**Review by codeliezel — APPROVED** [review]

**Pull request merged**